manifests/application.pp

```diff
@@ -49,30 +49,15 @@
     require        => [Class['certs::pulp_client'], Foreman::Rake['db:seed']],
   }
 
-  # We used to override permissions here so this matches it back to the packaging
-  file { '/usr/share/foreman/bundler.d/katello.rb':
-    ensure => file,
-    owner  => 'root',
-    group  => 'root',
-    mode   => '0644',
-  }
-
   include ::foreman::plugin::tasks
 
   Class['certs', 'certs::ca', 'certs::apache'] ~> Class['apache::service']
   Class['certs', 'certs::ca', 'certs::qpid'] ~> Class['foreman::plugin::tasks']
 
-  # Katello database seeding needs candlepin
-  package { $package_names:
-    ensure => installed,
-  } ->
-  file { "${::foreman::plugin_config_dir}/katello.yaml":
-    ensure  => file,
-    owner   => 'root',
-    group   => $::foreman::group,
-    mode    => '0640',
-    content => template('katello/katello.yaml.erb'),
-    notify  => [Class['foreman::service'], Foreman::Rake['db:seed'], Foreman::Rake['apipie:cache:index']],
```

Comment:
The seeding is a difference. With foreman we always do `database ~> plugin ~> service` where here we don't. That means it might get seeded with an invalid config if the package is already installed. Perhaps this is a bigger issue we need to address with katello's meta package anyway. For most plugins this isn't an issue because migrations typically don't use a config, but in Katello we do for Candlepin. That's why until now I've held off on this change.

Reply:
I know this passed in my testing. I can re-test this and report back.

Reply:
A fresh install and an upgrade test passed with this change. Perhaps we have just made the seeding more robust over time to not have the issue you were thinking of?